esphome/components/uptime/text_sensor/__init__.py

```diff
@@ -1,19 +1,55 @@
 import esphome.codegen as cg
-from esphome.components import text_sensor
+from esphome.components import text_sensor, time
 import esphome.config_validation as cv
-from esphome.const import ENTITY_CATEGORY_DIAGNOSTIC, ICON_TIMER
+from esphome.const import CONF_TIME_ID, ENTITY_CATEGORY_DIAGNOSTIC, ICON_TIMER
 
 uptime_ns = cg.esphome_ns.namespace("uptime")
-UptimeTextSensor = uptime_ns.class_(
-    "UptimeTextSensor", text_sensor.TextSensor, cg.PollingComponent
+UptimeSecondsTextSensor = uptime_ns.class_(
+    "UptimeSecondsTextSensor", text_sensor.TextSensor, cg.PollingComponent
 )
+UptimeTimestampTextSensor = uptime_ns.class_(
+    "UptimeTimestampTextSensor", text_sensor.TextSensor, cg.Component
+)
+
+
 CONFIG_SCHEMA = text_sensor.text_sensor_schema(
-    UptimeTextSensor,
+    UptimeSecondsTextSensor,
     icon=ICON_TIMER,
     entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
 ).extend(cv.polling_component_schema("30s"))
 
 
+CONFIG_SCHEMA = cv.typed_schema(
+    {
+        "seconds": text_sensor.text_sensor_schema(
```

Comment:
```suggestion
        "duration": text_sensor.text_sensor_schema(
```

Also consider defining constants `TYPE_TIMESTAMP` and `TYPE_DURATION` rather than embedding strings - not an absolute requirement when only used once, but is consistent with the style of other components.